ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java

```diff
@@ -135,69 +139,120 @@
     }
   }
 
-  private LineReader lineReader;
   private final ObjectMapper objectMapper;
   private final Map<String, CliSpecificCommand> cliSpecificCommands;
-
+  private final KsqlTerminal terminal;
+  private final RowCaptor rowCaptor;
   private OutputFormat outputFormat;
 
-  public Console(final OutputFormat outputFormat, final KsqlRestClient restClient) {
-    Objects.requireNonNull(
-        outputFormat,
-        "Must provide the terminal with a beginning output format"
-    );
-    Objects.requireNonNull(restClient, "Must provide the terminal with a REST client");
+  public interface RowCaptor {
```

Comment:
This interface is a bit of a hack to allow tests to capture rows. Personally, I would never add production code like this to support tests. However, in this instance the original code this PR refactors also has similar production code to support the tests and, in an attempt to keep this PR targeted, I've replicated that functionality via this interface.

To be picked up and refactored out another day...